src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicListener.cs

```diff
@@ -157,7 +159,8 @@ private void Dispose(bool disposing)
                 return;
             }
 
-            Stop();
+            // TODO: solve listener stopping in better way now that it receives STOP_COMPLETED event.
+            StopAsync().GetAwaiter().GetResult();
```

Comment:
We added this async stop functionality at the request of .NET. 😄 We were told it would make things easier for you!

Reply:
Yes, I have it in my TODO list and plan to address. This is just a hack to get it working and be able to move forward, I'm aware it's ugly as heck :smiley: